File: src/main/java/io/github/benjohnde/ankeader/parser/utils/ZipUtils.java
package io.github.benjohnde.ankeader.parser.utils;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;
import java.util.zip.ZipEntry;
import java.util.zip.ZipInputStream;

public class ZipUtils {
    public static void unzip(String filename, File toDirectory) throws IOException {
        byte[] buffer = new byte[1024];
        ZipInputStream zis = new ZipInputStream(new FileInputStream(filename));
        ZipEntry zipEntry = zis.getNextEntry();
        while (zipEntry != null) {
            String fileName = zipEntry.getName();
            File newFile = new File(toDirectory.getAbsolutePath(), fileName);
            FileOutputStream fos = new FileOutputStream(newFile);
            int len;
            while ((len = zis.read(buffer)) > 0) {
                fos.write(buffer, 0, len);
            }
            fos.close();
            zipEntry = zis.getNextEntry();
        }
        zis.closeEntry();
        zis.close();
    }
}

File: src/main/java/io/github/benjohnde/ankeader/parser/utils/ZipUtils.kt
package io.github.benjohnde.ankeader.parser.utils

import java.io.File
import java.io.FileInputStream
import java.io.FileOutputStream
import java.io.IOException
import java.util.zip.ZipEntry
import java.util.zip.ZipInputStream

object ZipUtils {
    @Throws(IOException::class)
    fun unzip(filename: String, toDirectory: File) {
        val buffer = ByteArray(1024)
        val zis = ZipInputStream(FileInputStream(filename))
        var zipEntry: ZipEntry? = zis.nextEntry
        while (zipEntry != null) {
            val fileName = zipEntry.name
            val newFile = File(toDirectory.absolutePath, fileName)
            val fos = FileOutputStream(newFile)
            var len: Int = zis.read(buffer)
            while (len > 0) {
                fos.write(buffer, 0, len)
                len = zis.read(buffer)
            }
            fos.close()
            zipEntry = zis.nextEntry
        }
        zis.closeEntry()
        zis.close()
    }
}
